panther_hardware_interfaces/src/panther_system.cpp

```diff
@@ -494,16 +495,29 @@ void PantherSystem::UpdateDriverState()
 {
   try {
     motors_controller_->UpdateDriversState();
-    UpdateDriverStateMsg();
     UpdateFlagErrors();
     UpdateDriverStateDataTimedOut();
   } catch (const std::runtime_error & e) {
-    RCLCPP_WARN_STREAM(
-      logger_, "An exception occurred while updating drivers states: " << e.what());
     roboteq_error_filter_->UpdateError(ErrorsFilterIds::READ_PDO_DRIVER_STATE, true);
+
+    RCLCPP_ERROR_STREAM_THROTTLE(
+      logger_, steady_clock_, 10000,
+      "An exception occurred while updating drivers states: " << e.what());
   }
 }
 
+void PantherSystem::UpdateEStopState()
+{
+  if (
+    motors_controller_->GetFrontData().IsHeartbeatTimeout() ||
+    motors_controller_->GetRearData().IsHeartbeatTimeout()) {
+    e_stop_->TriggerEStop();
+  }
+
+  const bool e_stop = e_stop_->ReadEStopState();
+  panther_system_ros_interface_->PublishEStopStateIfChanged(e_stop);
+}
```

Comment:
Won't this be trying to trigger e-stop on each read?

Reply:
Do you mean a scenario once we stay in a `heartbeat_timeout` error? Underneath we aim to disable a watchdog and do fast return once already disabled.